service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/history/SealedOrderHistoryService.java

```diff
@@ -40,6 +42,9 @@
     private final OrderCreationService orderCreationService;
     private final Time time;
 
+    // That should go in a similar strategy if more are coming
+    private final ManageOrdersClosedCaseFieldGenerator manageOrdersClosedCaseFieldGenerator;
```

Comment:
This might be because I'm not very used to the codebase, so feel free to disregard this. But reading the name of this class and method, it's not very clear to me what it does.

Reply:
If the name isn't clear to someone not familiar with the code then I think it might not be a good name but I am unsure as to what we could rename this class to

Reply:
What about something like "PrePopulator"?
It could be an interface that `ManageOrdersClosedCasePrePopulator` implements?

But again, I probably need to get used to these concepts, so feel free to disregard me on this one.